en/docs/how-tos/passing-enduser-attributes-to-the-backend-using-jwt.md

```diff
@@ -175,6 +175,69 @@ Outer `[jwtGeneratorConfig]` configuration contains the properties and the custo
 </tbody>
 </table>
 
+#### Adding Claims Retrieved from a remote endpoint
+
+If the user needs to retrieve user claims from a remote endpoint, the microgateway can be configured to support that 
+as well. The microgateway has an inbuilt implementation to get user claims from 
+`<WSO2 Key Manager URL>/keymanager-operations/user-info/claims/generate` endpoint. If you are willing to use the same 
+implementation, you can add the following configuration to the `micro-gw.conf` file.
+
+```toml
+[jwtGeneratorConfig.claimRetrieval]
+      retrieverImpl="org.wso2.micro.gateway.jwt.generator.MGWClaimRetrieverImpl"
+      [jwtGeneratorConfig.claimRetrieval.configuration]
+          # The User can provide any key-value pair here. These are the properties used by default.
+          # serverUrl = <Key Manager URL>
+          # username = <APIM Credentials Username>
+          # password = <APIM Credentials Password>
+```
+
+The `retrieveImpl` Key is used to provide the claim retriever Implementation. Since the value given here denotes an 
+inbuilt implementation, the user does not need to add any additional jar files to the project. Under the 
+`[jwtGeneratorConfig.claimRetrieval.configuration]`, the user can customize the properties as required. 
+
+In addition to that, it is required to enable claimRetrieval in `[[jwtTokenConfig]]` (per JWT issuer) and 
+`[keyManager]` Configurations based on user requirement. 
+
+``` toml tab="jwtTokenConfig"
+[[jwtTokenConfig]]
+    remoteUserClaimRetrievalEnabled = true
+```
+
+``` toml tab="keyManager"
+[keyManager]
+    remoteUserClaimRetrievalEnabled = true
+```
+
+!!! note
+    If you do not declare the `jwtGeneratorConfig.claimRetrieval.retrieverImpl` property in `micro-gw.conf`, 
+    the classloading operation will not happen within the microgateway. Hence, the claim retrieval process will 
+    not be executed.
+        
+
+There can be usecases where the user needs to come up with his/her own claim retrieval Implementation. In such 
+a scenario, the user needs to implement the specific class extended from 
+[AbstractMGWClaimRetriever](https://github.com/wso2/product-microgateway/blob/v3.2.0/components/micro-gateway-jwt-generator/src/main/java/org/wso2/micro/gateway/jwt/generator/AbstractMGWClaimRetriever.java) class.
+For that, the user can have the same maven dependency used for custom JWT Generation implementation. A sample 
+implementation is available [here](https://github.com/wso2/product-microgateway/tree/master/samples/sample-jwt-generator).
+In the end, you can add the corresponding jar file to the microgateway project's `/lib` directory prior to executing 
+`micro-gw build` command.
+
+```xml
```

Comment:
This is just hanging there without any reference to what this example is. So shall put it inside a sample box like this
```suggestion
```xml tab="Maven dependency"
```